js/components/util/files.ts

```diff
@@ -0,0 +1,176 @@
+import { includes } from 'js/types';
+import { initGamepad } from 'js/ui/gamepad';
+import {
+    DISK_FORMATS,
+    DriveNumber,
+    JSONDisk,
+    NIBBLE_FORMATS
+} from 'js/formats/types';
+import DiskII from 'js/cards/disk2';
+
+/**
+ * Routine to split a legacy hash into parts for disk loading
+ *
+ * @returns an padded array for 1 based indexing
+ */
+
+export const getHashParts = () => {
+    const parts = window.location.hash.match(/^#([^|]*)\|?(.*)$/) || ['', '', ''];
+    return ['', parts[1], parts[2]];
+};
+
+/**
+ * Update the location hash to reflect the current disk state, if possible
+ *
+ * @param parts a padded array with values starting at index 1
+ */
+
+export const setHashParts = (parts: string[]) => {
+    window.location.hash = `#${parts[1]}` + (parts[2] ? `|${parts[2]}` : '');
+};
+
+/**
+ * Local file loading routine. Allows a File object from a file
+ * selection form element to be loaded.
+ *
+ * @param disk2 Disk2 object
+ * @param number Drive number
+ * @param file Browser File object to load
+ * @returns true if successful
+ */
+
+export const loadLocalFile = async (
```

Comment:
I don't see why this method is `async`—it never `await`s anything, it just returns a `Promise` directly.